sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -170,7 +171,8 @@ private RequestBody toOkHttpRequestBody(BinaryData bodyContent, HttpHeaders head
 
         if (content instanceof ByteArrayContent
             || content instanceof StringContent
-            || content instanceof SerializableContent) {
+            || content instanceof SerializableContent
+            || content instanceof ByteBufferContent) {
```

Comment:
Good catch, long term I think we need to add some more metadata to `BinaryDataContent` where we don't need to do all this type checking as it breaks when new types are added.

Reply:
reading my thoughts - https://github.com/Azure/azure-sdk-for-java/issues/34501